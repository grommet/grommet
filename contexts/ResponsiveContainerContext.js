"use strict";

exports.__esModule = true;
exports.ResponsiveContainerContext = void 0;
var _react = _interopRequireDefault(require("react"));
function _interopRequireDefault(e) { return e && e.__esModule ? e : { "default": e }; }
var ResponsiveContainerContext = exports.ResponsiveContainerContext = /*#__PURE__*/_react["default"].createContext(false);