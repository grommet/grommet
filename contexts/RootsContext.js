"use strict";

exports.__esModule = true;
exports.useRoots = exports.RootsContext = void 0;
var _react = _interopRequireWildcard(require("react"));
function _interopRequireWildcard(e, t) { if ("function" == typeof WeakMap) var r = new WeakMap(), n = new WeakMap(); return (_interopRequireWildcard = function _interopRequireWildcard(e, t) { if (!t && e && e.__esModule) return e; var o, i, f = { __proto__: null, "default": e }; if (null === e || "object" != typeof e && "function" != typeof e) return f; if (o = t ? n : r) { if (o.has(e)) return o.get(e); o.set(e, f); } for (var _t in e) "default" !== _t && {}.hasOwnProperty.call(e, _t) && ((i = (o = Object.defineProperty) && Object.getOwnPropertyDescriptor(e, _t)) && (i.get || i.set) ? o(f, _t, i) : f[_t] = e[_t]); return f; })(e, t); }
// When toggling aria-hidden values, we only want to affect elements
// in the DOM that come from Grommet, so we track those elements in this
// context value. We also only want `trapFocus` to apply to the most recently
// opened Drop/Layer. See FocusedContainer.js
var RootsContext = exports.RootsContext = /*#__PURE__*/_react["default"].createContext(undefined);
var useRoots = exports.useRoots = function useRoots() {
  // If FocusedContainer is within Grommet React tree, there will be a value
  // from RootsContext. If not, set up a default value that can be passed
  // to drops that might open from within a drop.
  var defaultRoots = (0, _react.useRef)([]);
  var defaultValue = (0, _react.useMemo)(function () {
    return {
      roots: defaultRoots
    };
  }, []);
  var existingRoots = (0, _react.useContext)(RootsContext);
  return {
    contextValue: existingRoots || defaultValue,
    hasRoots: (existingRoots == null ? void 0 : existingRoots.roots) !== undefined
  };
};