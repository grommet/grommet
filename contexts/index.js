"use strict";

exports.__esModule = true;
var _AnalyticsContext = require("./AnalyticsContext");
Object.keys(_AnalyticsContext).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (key in exports && exports[key] === _AnalyticsContext[key]) return;
  exports[key] = _AnalyticsContext[key];
});
var _AnnounceContext = require("./AnnounceContext");
Object.keys(_AnnounceContext).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (key in exports && exports[key] === _AnnounceContext[key]) return;
  exports[key] = _AnnounceContext[key];
});
var _ContainerTargetContext = require("./ContainerTargetContext");
Object.keys(_ContainerTargetContext).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (key in exports && exports[key] === _ContainerTargetContext[key]) return;
  exports[key] = _ContainerTargetContext[key];
});
var _DataContext = require("./DataContext");
Object.keys(_DataContext).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (key in exports && exports[key] === _DataContext[key]) return;
  exports[key] = _DataContext[key];
});
var _ResponsiveContext = require("./ResponsiveContext");
Object.keys(_ResponsiveContext).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (key in exports && exports[key] === _ResponsiveContext[key]) return;
  exports[key] = _ResponsiveContext[key];
});
var _ResponsiveContainerContext = require("./ResponsiveContainerContext");
Object.keys(_ResponsiveContainerContext).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (key in exports && exports[key] === _ResponsiveContainerContext[key]) return;
  exports[key] = _ResponsiveContainerContext[key];
});
var _RootsContext = require("./RootsContext");
Object.keys(_RootsContext).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (key in exports && exports[key] === _RootsContext[key]) return;
  exports[key] = _RootsContext[key];
});
var _ThemeContext = require("./ThemeContext");
Object.keys(_ThemeContext).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (key in exports && exports[key] === _ThemeContext[key]) return;
  exports[key] = _ThemeContext[key];
});