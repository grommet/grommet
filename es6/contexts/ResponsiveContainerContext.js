import React from 'react';
export var ResponsiveContainerContext = /*#__PURE__*/React.createContext(false);