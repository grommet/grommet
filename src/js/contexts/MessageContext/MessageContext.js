import React from 'react';

import defaultMessages from '../../languages/default.json';

// options:
//   id: message id
//   messages: (optional) an object of message overrides
//   values: (optional) currently unused but in the future
//     will be an object with substitution values for
//     positional variables in the message text.
//   defaultMessage: (optional) default message to use if
//     the message isn't found elsewhere.
export const format = (options, messages) => {
  // Message id's are hierarchical. For the component-specific
  // message objects passed as options.messages, just use the last
  // component in the id for backwards compatibility.
  //
  // For overall messages passed to grommet, use the hierarchical
  // id. For that messages object, the messages are in an object
  // hierarchy by component, similar to how the theme works.
  //
  // Applications that typically keep their messages in flat
  // objects with a single key string per message can override
  // this format function to get the grommet messages from
  // their bundles that way and don't need to pass the messages
  // themselves in this property, just the format function.
  const idParts = options.id?.split('.') || [];
  // eslint-disable-next-line no-unsafe-optional-chaining
  const baseId = idParts[idParts?.length - 1];
  let messageObj = messages;
  idParts.forEach((idPart) => {
    if (typeof messageObj === 'object') {
      messageObj = messageObj[idPart];
    }
  });

  const message =
    (options.messages ? options.messages[baseId] : undefined) ||
    messageObj ||
    options.defaultMessage;

  const { values } = options;

  let newMessage = message;
  const tokens = message?.match(/\{(.+?)\}/g);
  tokens?.forEach((token) => {
    const names = token.substr(1, token.length - 2);
    const value = values[names];
    newMessage = newMessage.replace(token, value);
  });

  return values ? newMessage : message;
};

const defaultValue = {
  messages: defaultMessages,
  format: (options) => format(options, defaultMessages),
};

export const MessageContext = React.createContext(defaultValue);
