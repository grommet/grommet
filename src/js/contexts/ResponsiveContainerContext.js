import React from 'react';

export const ResponsiveContainerContext = React.createContext(false);
