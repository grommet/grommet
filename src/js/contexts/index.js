export * from './AnalyticsContext';
export * from './AnnounceContext';
export * from './ContainerTargetContext';
export * from './DataContext';
export * from './ResponsiveContext';
export * from './ResponsiveContainerContext';
export * from './RootsContext';
export * from './ThemeContext';
