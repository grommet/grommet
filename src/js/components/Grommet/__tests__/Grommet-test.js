import React from 'react';
import { render } from '@testing-library/react';
import 'jest-styled-components';

import { hpe as hpeTheme } from 'grommet-theme-hpe';

import { Grommet } from '..';
import { AnnounceContext } from '../../../contexts';
import { grommet } from '../../../themes/grommet';
import { MessageContext } from '../../../contexts/MessageContext';

const TestAnnouncer = ({ announce }) => {
  React.useEffect(() => announce('hello', 'assertive'));
  return <div>hi</div>;
};

describe('Grommet', () => {
  test('basic', () => {
    const { container } = render(<Grommet />);
    expect(container.firstChild).toMatchSnapshot();
  });

  test('grommet theme', () => {
    const { container } = render(<Grommet theme={grommet} />);
    expect(container.firstChild).toMatchSnapshot();
  });

  test('hpe theme', () => {
    const { container } = render(
      <Grommet theme={hpeTheme}>Grommet App</Grommet>,
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  test('themeMode', () => {
    const { container } = render(<Grommet theme={grommet} themeMode="dark" />);
    expect(container.firstChild).toMatchSnapshot();
  });

  test('cssVars', () => {
    const { container } = render(<Grommet cssVars>Grommet App</Grommet>);
    expect(container.firstChild).toMatchSnapshot();
  });

  test('full', () => {
    const { container } = render(<Grommet full>Grommet App</Grommet>);
    expect(container.firstChild).toMatchSnapshot();
  });

  test('full min', () => {
    const { container } = render(<Grommet full="min">Grommet App</Grommet>);
    expect(container.firstChild).toMatchSnapshot();
  });

  test('background', () => {
    const { container } = render(
      <Grommet full background="#0000ff">
        Grommet App
      </Grommet>,
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  test('announce', (done) => {
    const { container } = render(
      <Grommet>
        <AnnounceContext.Consumer>
          {(announce) => <TestAnnouncer announce={announce} />}
        </AnnounceContext.Consumer>
      </Grommet>,
    );

    expect(container.firstChild).toMatchSnapshot();

    // no style, no need for expectPortal
    expect(
      document.body.querySelector('#grommet-announcer[aria-live]'),
    ).toMatchSnapshot();

    setTimeout(() => {
      // should clear the aria-live container
      expect(
        document.body.querySelector('#grommet-announcer[aria-live]'),
      ).toMatchSnapshot();
      done();
    }, 600); // wait the aria-live container to clear
  });

  test('messages', () => {
    const { container } = render(
      <Grommet
        messages={{
          messages: {
            test: {
              label: 'My Label',
            },
          },
        }}
      >
        <MessageContext.Consumer>
          {({ format }) => format({ id: 'test.label' })}
        </MessageContext.Consumer>
      </Grommet>,
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  test('message format function', () => {
    const messages = {
      'test.label': 'My Label',
    };
    const { container } = render(
      <Grommet full background="#0000ff">
        Grommet App
      </Grommet>,

      <Grommet messages={{ format: (opts) => messages[opts.id] }}>
        <MessageContext.Consumer>
          {({ format }) => format({ id: 'test.label' })}
        </MessageContext.Consumer>
      </Grommet>,
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
