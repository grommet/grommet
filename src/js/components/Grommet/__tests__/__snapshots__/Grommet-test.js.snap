// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Grommet announce 1`] = `
.c0 {
  font-size: 18px;
  line-height: 24px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

<div
  class="c0"
>
  <div>
    hi
  </div>
</div>
`;

exports[`Grommet announce 2`] = `
<div
  aria-live="assertive"
  id="grommet-announcer"
  style="left: -100%; right: 100%; position: fixed; z-index: -1;"
>
  hello
</div>
`;

exports[`Grommet announce 3`] = `
<div
  aria-live="assertive"
  id="grommet-announcer"
  style="left: -100%; right: 100%; position: fixed; z-index: -1;"
/>
`;

exports[`Grommet background 1`] = `
.c0 {
  font-size: 18px;
  line-height: 24px;
  background-color: #0000ff;
  color: #f8f8f8;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
  width: 100vw;
  height: 100vh;
  overflow: auto;
}

<div
  class="c0"
>
  Grommet App
</div>
`;

exports[`Grommet basic 1`] = `
.c0 {
  font-size: 18px;
  line-height: 24px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

<div
  class="c0"
/>
`;

exports[`Grommet cssVars 1`] = `
.c0 {
  font-size: 18px;
  line-height: 24px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
  --active: rgba(221, 221, 221, 0.5);
  --active-background: background-contrast;
  --active-text: text-strong;
  --black: #000000;
  --brand: #7D4CDB;
  --focus: #6FFFB0;
  --graph-0: accent-1;
  --graph-1: neutral-1;
  --graph-2: neutral-2;
  --graph-3: neutral-3;
  --graph-4: neutral-4;
  --placeholder: #AAAAAA;
  --selected: brand;
  --selected-background: brand;
  --selected-text: text-strong;
  --white: #FFFFFF;
  --accent-1: #6FFFB0;
  --accent-2: #FD6FFF;
  --accent-3: #81FCED;
  --accent-4: #FFCA58;
  --dark-1: #333333;
  --dark-2: #555555;
  --dark-3: #777777;
  --dark-4: #999999;
  --dark-5: #999999;
  --dark-6: #999999;
  --light-1: #F8F8F8;
  --light-2: #F2F2F2;
  --light-3: #EDEDED;
  --light-4: #DADADA;
  --light-5: #DADADA;
  --light-6: #DADADA;
  --neutral-1: #00873D;
  --neutral-2: #3D138D;
  --neutral-3: #00739D;
  --neutral-4: #A2423D;
  --status-critical: #EB0000;
  --status-error: #B30000;
  --status-warning: #C27B00;
  --status-ok: #009E67;
  --status-unknown: #919191;
  --status-disabled: #CCCCCC;
}

<div
  class="c0"
>
  Grommet App
</div>
`;

exports[`Grommet full 1`] = `
.c0 {
  font-size: 18px;
  line-height: 24px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
  width: 100vw;
  height: 100vh;
  overflow: auto;
}

<div
  class="c0"
>
  Grommet App
</div>
`;

exports[`Grommet full min 1`] = `
.c0 {
  font-size: 18px;
  line-height: 24px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
  min-height: 100vh;
}

<div
  class="c0"
>
  Grommet App
</div>
`;

exports[`Grommet grommet theme 1`] = `
.c0 {
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Oxygen,Ubuntu,Cantarell,"Fira Sans","Droid Sans","Helvetica Neue",Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol";
  font-size: 18px;
  line-height: 24px;
  background-color: #ffffff;
  color: #444444;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

<div
  class="c0"
/>
`;

exports[`Grommet hpe theme 1`] = `
.c0 {
  font-family: 'Metric',Arial,sans-serif;
  font-size: 1.125rem;
  line-height: 1.5rem;
  background-color: #ffffff;
  color: #555555;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

<div
  class="c0"
>
  Grommet App
</div>
`;

exports[`Grommet message format function 1`] = `
.c0 {
  font-size: 18px;
  line-height: 24px;
  background-color: #0000ff;
  color: #f8f8f8;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
  width: 100vw;
  height: 100vh;
  overflow: auto;
}

<div
  class="c0"
>
  Grommet App
</div>
`;

exports[`Grommet messages 1`] = `
.c0 {
  font-size: 18px;
  line-height: 24px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

<div
  class="c0"
>
  My Label
</div>
`;

exports[`Grommet themeMode 1`] = `
.c0 {
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Oxygen,Ubuntu,Cantarell,"Fira Sans","Droid Sans","Helvetica Neue",Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol";
  font-size: 18px;
  line-height: 24px;
  background-color: #000000;
  color: #f8f8f8;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

<div
  class="c0"
/>
`;
